dv/management/commands/import_news.py

```diff
@@ -42,12 +41,24 @@ def _save(self, item):
             news.summary = item['summary']
             news.image = item['image'].replace('http://', 'https://')
             news.is_partnership = item['is_partnership'] == 'yes'
-            if item['project_id']:
-                news.project_id = item['project_id'].strip()
+            project_id = item.get("project_id")
+            if project_id:
+                project = Project.objects.filter(code=project_id).exists()
+                if project:
+                    news.project_id = project_id.strip()
+                else:
+                    self.stderr.write(f"Project code: {project_id} doesn't exist!")
             news.save()
 
-            for prg in item['programme_id'].split(', '):
-                news.programmes.add(prg.upper().strip())
+            item_programmes = item.get("programme_id")
+            if item_programmes:
+                item_programmes = item_programmes.split(", ")
```

Comment:
Rename to `programme_codes` or `programme_ids` and compute it in one go:

```python
item.get("programme_id", "").split(", ")
```

Reply:
Can't compute it in one go, because sometimes if the programme_id is `""` the split of it would be` ['']`